generators/lib/drupalProjectVersion.js

```diff
@@ -102,9 +102,20 @@ function init() {
 
   // Convert a semver version from '8.2.6' to a minor range '^8.2'.
   function toMinorRange(version) {
-    var regex = /^(\d+\.\d+)\.\d+/;
+    var regex = /^\d+\.\d+/;
     var range = version.match(regex);
-    return '^' + range[1];
+    if (range) {
+      return '^' + range[0];
+    }
+    else {
```

Comment:
The simpler code was introduced a few days ago. The changes allow it to more gracefully support not having a patch version, having a pre-release version, or in fact being handed a version such as 8.x, which is itself a minor range, and the kind we will now be getting from Octane.